corehq/apps/data_interfaces/tests/test_case_deduplication.py

```diff
@@ -98,6 +99,50 @@ def test_with_location_filter(self):
         self.assertTrue(retrieved_cases[0]['owner_id'] == location_id)
         self.assertTrue(retrieved_cases[1]['owner_id'] == location_id)
 
+    @patch.object(LocationType, 'commtrack_enabled', new_callable=PropertyMock, return_value=False)
+    @patch('corehq.apps.commtrack.models.sync_supply_point')
+    def test_with_child_location(self, sync_supply_mock, mock_commtrack_enabled):
+        loc_type = LocationType.objects.create(domain='test-domain', name='level1')
+        parent_location = SQLLocation.objects.create(
+            domain='test-domain', name='parent', location_type=loc_type, location_id='parent'
+        )
+        SQLLocation.objects.create(
+            domain='test-domain', name='child', location_type=loc_type, parent=parent_location, location_id='child'
+        )
+
+        cases = [
+            self.factory.create_case(case_name=case_name, update={'dob': dob}) for (case_name, dob) in [
+                ("Anakin Skywalker", "1977-03-25"),
+                ("Darth Vadar", "1977-03-25"),
+                ("Wannabe Anakin Skywalker", "1977-03-25"),
+                ("Wannabe Darth Vadar", "1977-03-25"),
+            ]
+        ]
+
+        rule = self.create_rule('test rule', cases[0].type)
+        definition = LocationFilterDefinition.objects.create(
+            location_id='parent',
+            include_child_locations=True,
+        )
+        criteria = CaseRuleCriteria(rule=rule)
+        criteria.definition = definition
+        criteria.save()
+
+        location_id = 'child'
+
+        # Only assign location id to first 2 cases, since we want only those two cases to be considered
+        cases[0].owner_id = location_id
+        cases[1].owner_id = location_id
+
+        self._prime_es_index(cases)
+
+        query = _get_es_filtered_case_query(self.domain, cases[0], rule.memoized_criteria)
+        retrieved_cases = query.run().hits
+
+        self.assertEqual(len(retrieved_cases), 2)
+        self.assertTrue(retrieved_cases[0]['owner_id'] == location_id)
+        self.assertTrue(retrieved_cases[1]['owner_id'] == location_id)
```

Comment:
nit: this could be checked with a single assertion that would have more informative output on failure.
```suggestion
        self.assertEqual([rc['owner_id'] for rc in retrieved_cases], [location_id, location_id])
```